jira/client.py

```diff
@@ -2076,6 +2076,76 @@ def _check_createmeta_issuetypes(self) -> None:
+        self,
+        projectIdOrKey: str | int,
+        startAt: int = 0,
+        maxResults: int = 50,
+    ) -> dict[str, Any]:
+        """Get the issue types metadata for a given project, required to create issues.
+        This API was introduced in JIRA Server / DC 8.4 as a replacement for the more general purpose API 'createmeta'.
+        For details see: https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html
+        Args:
+            projectIdOrKey (Union[str, int]): id or key of the project for which to get the metadata.
+            startAt (int): Index of the first issue to return. (Default: ``0``)
+            maxResults (int): Maximum number of issues to return.
+              Total number of results is available in the ``total`` attribute of the returned :class:`ResultList`.
+              If maxResults evaluates to False, it will try to get all issues in batches. (Default: ``50``)
+
+        Returns:
+            Dict[str, Any]
+        """
+        warnings.warn(
+            "'createmeta_issuetypes' is deprected and will be removed in future releases."
+            "Use 'project_issue_types' instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        self._check_createmeta_issuetypes()
+        return self._get_json(
+            f"issue/createmeta/{projectIdOrKey}/issuetypes",
+            params={
+                "startAt": startAt,
+                "maxResults": maxResults,
+            },
+        )
+
+    def createmeta_fieldtypes(
+        self,
+        projectIdOrKey: str | int,
+        issueTypeId: str | int,
+        startAt: int = 0,
+        maxResults: int = 50,
+    ) -> dict[str, Any]:
+        """Get the field metadata for a given project and issue type, required to create issues.
+        This API was introduced in JIRA Server / DC 8.4 as a replacement for the more general purpose API 'createmeta'.
+        For details see: https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html
+        Args:
+            projectIdOrKey (Union[str, int]): id or key of the project for which to get the metadata.
+            issueTypeId (Union[str, int]): id of the issue type for which to get the metadata.
+            startAt (int): Index of the first issue to return. (Default: ``0``)
+            maxResults (int): Maximum number of issues to return.
+              Total number of results is available in the ``total`` attribute of the returned :class:`ResultList`.
+              If maxResults evaluates to False, it will try to get all issues in batches. (Default: ``50``)
+
+        Returns:
+            Dict[str, Any]
+        """
+        warnings.warn(
+            "'createmeta_fieldtypes' is deprected and will be removed in future releases."
+            "Use 'project_issue_fields' instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        self._check_createmeta_issuetypes()
```

Comment:
```suggestion
        """Get the field metadata for a given project and issue type, required to create issues.

        .. deprecated:: 3.6.0
            Use :func:`project_issue_fields` instead.
        
        This API was introduced in JIRA Server / DC 8.4 as a replacement for the more general purpose API 'createmeta'.
        For details see: https://confluence.atlassian.com/jiracore/createmeta-rest-endpoint-to-be-removed-975040986.html

        Args:
            projectIdOrKey (Union[str, int]): id or key of the project for which to get the metadata.
            issueTypeId (Union[str, int]): id of the issue type for which to get the metadata.
            startAt (int): Index of the first issue to return. (Default: ``0``)
            maxResults (int): Maximum number of issues to return.
              Total number of results is available in the ``total`` attribute of the returned :class:`ResultList`.
              If maxResults evaluates to False, it will try to get all issues in batches. (Default: ``50``)

        Returns:
            Dict[str, Any]
        """
        warnings.warn(
            "'createmeta_fieldtypes' is deprecated and will be removed in future releases."
            "Use 'project_issue_fields' instead.",
            DeprecationWarning,
            stacklevel=2,
        )
        self._check_createmeta_issuetypes()
```